ue5osc/__init__.py

```diff
@@ -1,21 +1,103 @@
 from pythonosc import udp_client
+from time import sleep
+
+from ue5osc.osc_dispatcher import DispatchHandler
+from pythonosc.osc_server import BlockingOSCUDPServer
+from PIL import Image
+
+import matplotlib.pyplot as plt
 
 
 class OSCSender:
-    def __init__(self, ip: str, port: int):
+    """Handles interaction between the UE5 environment and the a program."""
+
+    # Initializing the OSC communicator
+    def __init__(self, ip: str, port: int, directory: str):
+        self.path = directory
+        self.location_handler = DispatchHandler()
         self.client = udp_client.SimpleUDPClient(ip, port)
+        self.server = BlockingOSCUDPServer((ip, 7001), self.location_handler.dispatcher)
+        self.imgnumber = 0
+
+    # Starts the OSC server
+    def start_server(self):
+        self.server.serve_forever()

```

Comment:
Any reason we should star the server in the constructor?

Reply:
We are a bit confused as to what you exactly are asking. Are you referring to moving self.server.serve_forever() into the constructor?

Reply:
Sorry for my typing mistakes!

Yes, that is exactly what I mean. Any reason that we shouldn't call `self.server.serve_forever()` in the constructor?

Reply:
No, I moved it into the constructor, but by doing so I also had to make a function to shutdown the server and the thread as well.

Reply:
For some reason calling it in the constructor class makes it so it doesn't want to run so we decided to keep it as is. The error is below:

C:\Users\fxmn2022\Documents\ue5osc>python demo.py C:\Users\fxmn2022\Pictures\Simulation
Traceback (most recent call last):
  File "C:\Users\fxmn2022\Documents\ue5osc\demo.py", line 54, in <module>
    main()
  File "C:\Users\fxmn2022\Documents\ue5osc\demo.py", line 21, in main
    osc_sender = ue5osc.Communicator(ip, client_port, server_port, args.path)
  File "C:\Users\fxmn2022\Documents\ue5osc\ue5osc\__init__.py", line 26, in __init__    
    self.server_thread = threading.Thread(target=self.server.serve_forever())
  File "C:\Users\fxmn2022\AppData\Local\mambaforge\lib\socketserver.py", line 232, in serve_forever
    ready = selector.select(poll_interval)
  File "C:\Users\fxmn2022\AppData\Local\mambaforge\lib\selectors.py", line 324, in select
    r, w, _ = self._select(self._readers, self._writers, [], timeout)
  File "C:\Users\fxmn2022\AppData\Local\mambaforge\lib\selectors.py", line 315, in _select
    r, w, x = select.select(r, w, w, timeout)